src/plugins/intel_gpu/src/graph/impls/ocl/slice.cpp

```diff
@@ -40,104 +39,152 @@ std::vector<std::int32_t> extractIntegerData(const data_node& node, const stream
+        kernel_arguments_data args;
+
+        const SliceKernelRefNeededInputs inputs = SliceKernelRefNeededInputs::Create(*instance.node);
+
+        for (auto idx : inputs.GetNeededInputIndexes()) {
+            args.inputs.push_back(instance.input_memory_ptr(idx));
+        }
+
+        for (size_t i = 0; i < instance.outputs_memory_count(); i++) {
+            args.outputs.push_back(instance.output_memory_ptr(i));
+        }
+
+        args.shape_info = instance.shape_info_memory_ptr();
+        return args;
+    }
+
     static std::unique_ptr<primitive_impl> create(const slice_node& arg, const kernel_impl_params& impl_param) {
-        auto params = get_default_params<kernel_selector::slice_params>(impl_param);
-        const auto& inputs = arg.get_dependencies();
-        const stream& stream = arg.get_program().get_stream();
-        auto start_elts = extractIntegerData(inputs[InputIndices::kStart].first->as<data>(), stream);
-        auto end_elts = extractIntegerData(inputs[InputIndices::kEnd].first->as<data>(), stream);
-        auto step_elts = extractIntegerData(inputs[InputIndices::kStep].first->as<data>(), stream);
-        auto data_shape = extractShape(params.inputs[0]);
-        std::vector<std::int32_t> axes(data_shape.size());
-        if (inputs.size() == InputIndices::kInputsNum)
-            axes = extractIntegerData(inputs[InputIndices::kAxes].first->as<data>(), stream);
-        else
-            std::iota(axes.begin(), axes.end(), 0);
-        std::vector<std::int32_t> selected_start(data_shape.size(), 0);
-        std::vector<std::int32_t> selected_step(data_shape.size(), 1);
-        std::vector<std::int32_t> selected_end(data_shape);
-        for (size_t axis = 0; axis < axes.size(); axis++) {
-            auto transformed_axe = axes[axis] < 0 ? data_shape.size() + axes[axis] : axes[axis];
-            auto start = start_elts[axis];
-            auto end = end_elts[axis];
-            auto dim_size = data_shape[transformed_axe];
-            selected_start[transformed_axe] = std::max(std::min(start < 0 ? dim_size + start : start, dim_size - 1), 0);
-            selected_end[transformed_axe] = std::max(std::min(end < 0 ? dim_size + end : end, dim_size - 1), 0);
-            selected_step[transformed_axe] = step_elts[axis];
+        auto params = get_default_params<kernel_selector::slice_params>(impl_param, impl_param.is_dynamic());
+
+        OPENVINO_ASSERT(PrepareInput(arg,
+                                     SliceKernelRefNeededInputs::kStart,
+                                     params.compile_time_start,
+                                     params.start_data_type,
+                                     params.inputs),
+                        "[GPU][SliceOp]: Slice op definition does not contain start input!");
+
+        // NOTE: Stop input is not used by the slice kernel, as this information
+        // is implicitely passed with output shape.
+
+        OPENVINO_ASSERT(PrepareInput(arg,
```

Comment:
As this `PrepareInput` function is doing something, not only checking, I think it would be better to move it outside of OPENVINO_ASSERT, just to improve readability a little bit:
```
bool status = PrepareInput(..);
OPENVINO_ASSERT(success, "msg");

status = PrepareInput(..);
OPENVINO_ASSERT(success, "msg");
...
```

Reply:
Fixed, but in different way: assert is removed completely, since by def step and start may not be set at all and then default values should be applied.